docs/guides/migrations/guide.rst

```diff
@@ -0,0 +1,1298 @@
+    {true}
+
+Looks good! And now we can simply remove ``multi friends: User;`` 
+from our schema and do a final migration.
+
+Migration internals
+===================
+
+We've now reached the most optional part of the migrations tutorial, 
+but an interesting one for those curious about what goes on behind 
+the scenes during a migration.
+
+Migrations in EdgeDB before the advent of the EdgeDB Project flow 
+were still automated, but required more manual work if you didn't 
+want to accept all of the suggestions provided by the server. This 
+process is in fact still used to migrate even today; the CLI just 
+facilitates it by making it easy to respond to the generated suggestions.
+
+They took place `inside a transaction <transaction_>`_ handled by 
+the user that essentially went like this: 
+
+.. code-block::
+
+    db> start migration to { <your schema goes here> };
+
+This starts the migration, after which the quickest process was to 
+type ``populate migration`` to accept the statements suggested by 
+the server, and then ``commit migration`` to finish the process.
+
+Now, there is another option besides simply typing ``populate migration`` 
+that allows you to look at and handle the suggestions every step of 
+the way (in the same way the CLI does today), and this is what we 
+are going to have some fun with. You can see `the original RFC <rfc_>`_ 
+for this if you are curious.
+
+It is *very* finicky compared to the CLI, resulting in a failed transaction 
+if any step along the way is different from the expected behavior, 
+but is an entertaining challenge to try to get right if you want to 
+truly understand how migrations work in EdgeDB.
+
+This process requires looking at the server's proposed solutions every 
+step of the way, and these steps are best seen in JSON format. Let's 
+first set the CLI to make the format nicely readable with this command:
+
+.. code-block:: edgeql-repl
+
+    db> \set output format json-pretty
+
+We will begin with the same simple schema used in the previous examples:
+
+.. code-block:: sdl
+
+    module default {
+      type User {
+        name: str;
+      }
+    }
+
+And, as before, we will make a somewhat ambiguous change by changing 
+``name`` to ``nam``.
+
+.. code-block:: sdl
+
+    module default {
+      type User {
+        nam: str;
+      }
+    }
+
+Now we can start a migration by pasting this inside a ``start migration 
+to {};`` block:
+
+.. code-block:: edgeql-repl
+
+    db> start migration to {
+    ...   module default {
+    ...     type User {
+    ...       name: str;
+    ...     }
+    ...   }
+    ... };
+
+You should get the output ``OK: START MIGRATION``, followed by a prompt 
+that ends with ``[tx]`` to show that we are inside of a transaction. 
+Anything we do here will have no effect on the current registered 
+schema until we finally commit the migration.
+
+So now what do we do? We could simply type ``populate migration`` 
```

Comment:
I think we've skipped a step here. The `describe migration` output suggests we have migrated to the new property name `nam`, but I don't think we've actually performed that migration yet. We need to first complete the initial migration and then start a new one with the new property name before we can get to this point.

Reply:
Ah, typo there. I've rephrased a bit while fixing it: https://github.com/edgedb/edgedb/pull/6656/commits/e6b19a539d1d3d1377d7203203d502ad39bd1300